bluepyopt/ephys/create_hoc.py

```diff
@@ -33,53 +49,106 @@
     'myelinated']
 
 
-def _generate_channels_by_location(mechs, location_order):
+def _generate_channels_by_location(mechs, location_order, loc_desc):
     """Create a OrderedDictionary of all channel mechs for hoc template."""
     channels = OrderedDict((location, []) for location in location_order)
+    point_channels = OrderedDict((location, []) for location in location_order)
     for mech in mechs:
         name = mech.suffix
         for location in mech.locations:
-            # TODO this is dangerous, implicitely assumes type of location
-            channels[location.seclist_name].append(name)
-    return channels
+            if isinstance(mech, NrnMODPointProcessMechanism):
+                point_channels[loc_desc(location, mech)].append(mech)
+            else:
+                channels[loc_desc(location, mech)].append(name)
+    return channels, point_channels
 
 
 def _generate_reinitrng(mechs):
     """Create re_init_rng function"""
 
+    for mech in mechs:
+        if isinstance(mech, NrnMODPointProcessMechanism):
+            raise NotImplementedError(
+                'HOC generation for models with point process mechanisms'
+                ' is not yet supported.')
+
     reinitrng_hoc_blocks = ''
 
     for mech in mechs:
         reinitrng_hoc_blocks += mech.generate_reinitrng_hoc_block()
 
-    reinitrng_content = mechanisms.NrnMODMechanism.hash_hoc_string
+    reinitrng_content = NrnMODMechanism.hash_hoc_string
 
-    reinitrng_content += mechanisms.NrnMODMechanism.reinitrng_hoc_string % {
+    reinitrng_content += NrnMODMechanism.reinitrng_hoc_string % {
         'reinitrng_hoc_blocks': reinitrng_hoc_blocks}
 
     return reinitrng_content
 
 
-def _generate_parameters(parameters):
+def _range_exprs_to_hoc(range_params):
+    """Process raw range parameters to hoc strings"""
+
+    ret = []
+    for param in range_params:
+        value = param.value_scaler.inst_distribution
+        value = re.sub(r'math\.', '', value)
+        value = re.sub('{distance}', FLOAT_FORMAT, value)
+        value = re.sub('{value}', format_float(param.value), value)
+        ret.append(Range(param.location, param.name, value))
+    return ret
+
+
+def _loc_desc(location, param_or_mech):
+    """Generate Neuron location description for HOC template"""
+
+    if isinstance(param_or_mech, Mechanism):
+        if isinstance(param_or_mech, NrnMODMechanism):
+            if isinstance(location, NrnSeclistLocation):
+                return location.seclist_name
```

Comment:
The `isinstance` for tuples is a logical or, but here we need a logical and. Also, note the changing first argument. I think for now the nested if-strategy seems to be easier to read and maintain as which exact objects (locations/parameters/mechs) are supported in HOC is nowhere else really documented.

Reply:
Ah, I see. Ok. In this case, since `NrnMODMechanism` is a `Mechanism`, we only need two of the three checks.


`class NrnMODMechanism(Mechanism, serializer.DictMixin):`

Reply:
Then it's not explicit that the top-level distinction is on whether `param_or_mech` is a `Mechanism` or `Parameter`. Furthermore, there's already now more than one type of `Mechanism` and there may be more in the future. Flattening the nested ifs here will make the code harder to maintain in my opinion.